### PR TITLE
Metrics app Milestone 4

### PR DESCRIPTION
**Pull request by ambrose-kibet**
## In this milestone we:
- [x] Used redux to fetch and persist data from the api
- [x] Created the UI for the metrics app.
- [x] Added test for the metrics app

### CODE REVIEW

File: src/pages/Homepage.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        <article className="home-info">
+          <h2> Crypto-Currencies </h2>
+        </article>
+      </div>
+      <h3>Stats per rank</h3>
+      <div className="home-container">
+        {curencies.map(({
+          name, rank, id, symbol,
+        }) => (
+          <SingleCurrency
```
Comment (Ismailco) [inline src/pages/Homepage.js]: - [x] The list of coins in the `Home page` can be filtered using given parameters (like the name of the coin for example), kindly make sure to add a search area on the home page that can filter the list of coins and display only relevant ones when searching

File: src/pages/Homepage.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const Homepage = () => {
+  const { curencies } = useSelector((state) => state.currencies);
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    dispatch(getCrypto());
+  }, [dispatch]);
+
+  return (
+    <section className="section-container">
```
Comment (Ismailco) [inline src/pages/Homepage.js:17]: - [x] Kindly make sure to follow the provided design, you can use the `opacity` property to add a different color shad for every second column on the Home page or just add two different shades of the same color to style the coins cards

|Your app|Provided design|
|-|-|
|Screenshot_20230322_163017|Screenshot_20230322_163141|

File: src/pages/DetailsPage.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    <section className="section-container">
+      <HeroComponent info="single crypto info" />
+      <div className="home-intros space">
+        <div className="crypto-img-container">
+          <img
+            src={`https://coinicons-api.vercel.app/api/icon/${singleCrypto?.symbol}`}
+            alt={singleCrypto?.name}
+            className="crpto-img"
+          />
+        </div>
```
Comment (Ismailco) [inline src/pages/DetailsPage.js:27]: - [x] Same here, kindly make sure to follow the provided design, you can use the `opacity` property to add a different color shad for every second column on the Detail page or just add two different shades of the same color to style the information rows

|Your app|Provide design|
|-|-|
|Screenshot_20230322_163419|Screenshot_20230322_163152|

**Review by Ismailco — CHANGES_REQUESTED** [review]
#### Hi @ambrose-kibet, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️ Clean code
✔️ Implement testing
✔️ No linter errors
✔️ Used Gitflow
✔️ Work documented professionally

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by mwanawabangona — APPROVED** [review]
Hi @ambrose-kibet 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job implementing the requested changes!! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by amejid — COMMENTED** [review]
Hi @ambrose-kibet, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights
- Good test coverage :heavy_check_mark: 
- Functionalities working as expected :heavy_check_mark: 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**